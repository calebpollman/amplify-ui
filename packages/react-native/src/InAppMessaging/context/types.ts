/*
 * Copyright 2017-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import { ReactNode, ReactElement } from 'react';
import { InAppMessage } from '@aws-amplify/notifications';

import {
  BannerMessageProps,
  CarouselMessageProps,
  FullScreenMessageProps,
  ModalMessageProps,
  InAppMessageComponentStyles,
} from '../components';

export type InAppMessageComponents = {
  BannerMessage?: (props: BannerMessageProps) => ReactElement;
  CarouselMessage?: (props: CarouselMessageProps) => ReactElement;
  FullScreenMessage?: (props: FullScreenMessageProps) => ReactElement;
  ModalMessage?: (props: ModalMessageProps) => ReactElement;
};

export type InAppMessagingContextType = {
  clearInAppMessage: () => void;
  components: InAppMessageComponents;
  displayInAppMessage: (inAppMessage: InAppMessage) => void;
  inAppMessage: InAppMessage;
  style: InAppMessageComponentStyles;
};

export type InAppMessagingProviderProps = {
  children: ReactNode;
  components?: InAppMessageComponents;
  style?: InAppMessageComponentStyles;
};
