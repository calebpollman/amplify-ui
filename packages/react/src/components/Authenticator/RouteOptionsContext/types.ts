import { ReactNode } from 'react';

export type AuthenticatorRoute =
  | 'confirmResetPassword'
  | 'confirmSignIn'
  | 'confirmSignUp'
  | 'confirmVerifyUser'
  | 'forceNewPassword'
  | 'resetPassword'
  | 'setupTOTP'
  | 'signIn'
  | 'signUp'
  | 'verifyUser';

type OverrideComponent = <P>(props?: P) => JSX.Element;

interface RouteComponents {
  Component: OverrideComponent;
  FormFields?: OverrideComponent;
  Header?: OverrideComponent;
  Footer?: OverrideComponent;
}

// TODO: move other configuration options here, e.g. formFields
/**
 * Default options for an Authenticator Route.
 */
export type RouteOptionsDefault = Pick<RouteComponents, 'Component'>;

/**
 * Default options for all Authenticator Routes.
 */
export type RouteOptionsDefaults = {
  [K in AuthenticatorRoute]: RouteOptionsDefault;
};

/**
 * Override options for an Authenticator Route. Accepts either an override component or an options object.
 */
export type RouteOptionsOverride =
  | Omit<RouteComponents, 'Component'>
  | OverrideComponent;

/**
 * Default options for all Authenticator Routes.
 */
export type RouteOptionsOverrides = {
  [K in AuthenticatorRoute]?: RouteOptionsOverride;
};

export interface RouteOptionsProviderProps
  extends Omit<RouteComponents, 'Component'> {
  children: ReactNode;
  routeOptionsDefaults: RouteOptionsDefaults;
  routeOptionsOverrides: RouteOptionsOverrides;
}

// TODO: expand to include route specific items
export type RouteOptions = RouteComponents;
export type Routes = {
  [K in AuthenticatorRoute]: RouteComponents;
};

export interface RouteOptionsContextType
  extends Omit<RouteComponents, 'Component'> {
  routes: Routes;
}
