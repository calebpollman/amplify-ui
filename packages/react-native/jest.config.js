module.exports = {
  preset: 'react-native',
  modulePathIgnorePatterns: ['<rootDir>/dist/'],
  collectCoverageFrom: [
    '<rootDir>/src/**/*.{js,jsx,ts,tsx}',
    '!<rootDir>/src/**/*{c,C}onstants.ts',
  ],
  coverageThreshold: {
    global: {
      branches: 16,
      functions: 9,
      lines: 16,
      statements: 16,
    },
  },
};
