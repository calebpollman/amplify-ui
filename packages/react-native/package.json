{
  "name": "@aws-amplify/ui-react-native",
  "version": "0.1.0",
  "main": "dist/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/index.d.ts",
  "license": "Apache-2.0",
  "scripts": {
    "prebuild": "rimraf dist",
    "build": "yarn build:ts && cp -a src/assets dist",
    "build:ts": "ttsc --project tsconfig.json",
    "dev": "yarn build:ts --watch",
    "clean": "rimraf dist",
    "lint": "ttsc --noEmit && eslint src",
    "test": "yarn test:unit",
    "test:watch": "yarn test:unit:watch",
    "test:unit": "jest --config jest.config.js",
    "test:unit:watch": "jest --watch",
    "test:verbose": "yarn test -- --verbose"
  },
  "devDependencies": {
    "@babel/cli": "^7.15.7",
    "@babel/core": "7.15.5",
    "@babel/preset-env": "7.15.6",
    "@testing-library/react-hooks": "^7.0.2",
    "@types/react": "^16.13.1",
    "@types/react-native": "^0.63.45",
    "@types/react-test-renderer": "^17.0.1",
    "@typescript-eslint/eslint-plugin": "^4.32.0",
    "@typescript-eslint/parser": "^4.32.0",
    "@zerollup/ts-transform-paths": "^1.7.18",
    "babel-jest": "^27.2.4",
    "eslint": "^7.32.0",
    "eslint-config-airbnb": "^18.2.1",
    "eslint-config-airbnb-typescript": "^14.0.0",
    "eslint-config-prettier": "^8.3.0",
    "eslint-plugin-import": "^2.24.2",
    "eslint-plugin-jest": "^24.5.0",
    "eslint-plugin-jsx-a11y": "^6.4.1",
    "eslint-plugin-prettier": "^4.0.0",
    "eslint-plugin-react": "^7.26.0",
    "eslint-plugin-react-hooks": "^4.2.0",
    "metro-react-native-babel-preset": "^0.66.2",
    "prettier": "2.4.1",
    "react": "latest",
    "react-native": "^0.66.4",
    "react-test-renderer": "^17.0.2",
    "rimraf": "^3.0.2",
    "ts-jest": "^27.1.3",
    "ttypescript": "^1.5.13"
  },
  "dependencies": {
    "@aws-amplify/ui": "3.0.7",
    "@aws-amplify/ui-react-core": "0.1.0",
    "@aws-amplify/notifications": "in-app-messaging"
  },
  "peerDependencies": {
    "aws-amplify": "in-app-messaging",
    "react": ">= 16.8.0",
    "react-native": ">= 0.63.4"
  },
  "resolutions": {
    "@types/react": "^16.13.1"
  }
}
