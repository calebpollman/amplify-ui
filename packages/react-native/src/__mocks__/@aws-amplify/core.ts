/*
 * Copyright 2017-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

// export unmocked core modules
export * from '@aws-amplify/core';

// mock log functions
const mockDebug = jest.fn();
const mockError = jest.fn();
const mockInfo = jest.fn();
const mockWarn = jest.fn();
const mockLog = jest.fn();

const mockVerbose = jest.fn();
const mockAddPluggable = jest.fn();
const mockListPluggables = jest.fn();

// mock logger
const ConsoleLogger = jest.fn().mockImplementation(() => {
  return {
    addPluggable: mockAddPluggable,
    debug: mockDebug,
    error: mockError,
    info: mockInfo,
    listPluggables: mockListPluggables,
    log: mockLog,
    verbose: mockVerbose,
    warn: mockWarn,
  };
});

export { ConsoleLogger };
