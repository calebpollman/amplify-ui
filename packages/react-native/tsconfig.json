{
  "compilerOptions": {
    "allowSyntheticDefaultImports": true,
    "baseUrl": ".",
    "declaration": true,
    "downlevelIteration": true,
    "esModuleInterop": true,
    "jsx": "react-native",
    "noImplicitAny": false,
    "module": "es2015",
    "moduleResolution": "node",
    "noEmitOnError": true,
    "outDir": "./dist",
    "plugins": [
      {
        "transform": "@zerollup/ts-transform-paths",
        "exclude": ["*"]
      }
    ],
    "paths": { "@ui": ["src/ui"] },
    "skipLibCheck": true,
    "target": "es5"
  },
  "include": ["src/**/*.ts", "src/**/*.tsx"],
  "exclude": ["node_modules", "jest.config.js"]
}
