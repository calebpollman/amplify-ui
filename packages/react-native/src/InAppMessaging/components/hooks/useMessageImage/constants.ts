/*
 * Copyright 2017-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import { Dimensions } from 'react-native';
import { ImageDimensions } from './types';

// as images are not expected to be responsive to orientation changes get screen dimensions at app start
const SCREEN_DIMENSIONS = Dimensions.get('screen');

// compare screen width and height and assign the lesser of the two as the base screen dimension
const BASE_SCREEN_DIMENSION =
  SCREEN_DIMENSIONS.width < SCREEN_DIMENSIONS.height
    ? SCREEN_DIMENSIONS.width
    : SCREEN_DIMENSIONS.height;

// base size that message images should fill
// - all banner message images should fill 20 percent of the base screen dimension
// - all other components should fill 60 percent of the base screen dimension
export const BANNER_IMAGE_SCREEN_MULTIPLIER = 0.2;
export const CAROUSEL_IMAGE_SCREEN_MULTIPLIER = 0.6;
export const FULL_SCREEN_IMAGE_SCREEN_MULTIPLIER = 0.6;
export const MODAL_IMAGE_SCREEN_MULTIPLIER = 0.6;

export const BANNER_IMAGE_SCREEN_SIZE =
  BANNER_IMAGE_SCREEN_MULTIPLIER * BASE_SCREEN_DIMENSION;
export const CAROUSEL_IMAGE_SCREEN_SIZE =
  CAROUSEL_IMAGE_SCREEN_MULTIPLIER * BASE_SCREEN_DIMENSION;
export const FULL_SCREEN_IMAGE_SCREEN_SIZE =
  FULL_SCREEN_IMAGE_SCREEN_MULTIPLIER * BASE_SCREEN_DIMENSION;
export const MODAL_IMAGE_SCREEN_SIZE =
  MODAL_IMAGE_SCREEN_MULTIPLIER * BASE_SCREEN_DIMENSION;

export const INITIAL_IMAGE_DIMENSIONS: ImageDimensions = {
  height: null,
  width: null,
};
