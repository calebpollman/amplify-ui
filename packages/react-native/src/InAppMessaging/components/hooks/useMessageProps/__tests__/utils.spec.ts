/*
 * Copyright 2017-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import { PressableStateCallbackType, StyleProp, ViewStyle } from 'react-native';
import { InAppMessageTextAlign } from '@aws-amplify/notifications';

import { BUTTON_PRESSED_OPACITY } from '../../../constants';
import {
  InAppMessageComponentBaseProps,
  InAppMessageComponentBaseStyle,
} from '../../../types';
import { StyleParams } from '../types';

import {
  getComponentButtonStyle,
  getContainerAndWrapperStyle,
  getMessageStyle,
  getMessageStyleProps,
} from '../utils';

type ResolveContainerStyle = {
  container: (state?: PressableStateCallbackType) => StyleProp<ViewStyle>;
};

const EMPTY_STYLE = Object.freeze({});

describe('getComponentButtonStyle', () => {
  const pressedOpacity = { opacity: BUTTON_PRESSED_OPACITY };

  it.each(['primaryButton' as const, 'secondaryButton' as const])(
    'returns the expected output in the happy path for a %s',
    (buttonType) => {
      const defaultStyle = {
        buttonContainer: { backgroundColor: 'white' },
        buttonText: { color: 'red' },
      } as InAppMessageComponentBaseStyle;
      const messageStyle = {
        [buttonType]: {
          backgroundColor: 'maroon',
          borderRadius: 4,
          color: 'teal',
        },
      };
      const overrideStyle = {
        [buttonType]: {
          container: { backgroundColor: 'pink' },
          text: { color: 'black' },
        },
      };
      const styleParams = { defaultStyle, messageStyle, overrideStyle };

      const expectedContainerPressedStyle = [
        pressedOpacity,
        { backgroundColor: 'white' },
        { backgroundColor: 'maroon', borderRadius: 4 },
        { backgroundColor: 'pink' },
      ];
      const expectedContainerUnpressedStyle = [
        EMPTY_STYLE,
        { backgroundColor: 'white' },
        { backgroundColor: 'maroon', borderRadius: 4 },
        { backgroundColor: 'pink' },
      ];

      const expectedTextStyle = [
        { color: 'red' },
        { color: 'teal' },
        { color: 'black' },
      ];

      const output = getComponentButtonStyle({ styleParams, buttonType });

      const containerPressedStyle = (output as ResolveContainerStyle).container(
        { pressed: true }
      );
      const containerUnpressedStyle = (
        output as ResolveContainerStyle
      ).container({ pressed: false });

      expect(containerPressedStyle).toStrictEqual(
        expectedContainerPressedStyle
      );
      expect(containerUnpressedStyle).toStrictEqual(
        expectedContainerUnpressedStyle
      );

      expect(output.text).toStrictEqual(expectedTextStyle);
    }
  );

  it.each([{}, null])(
    'correctly handles a value of %s passed as messageStyle.primaryButton',
    (messageStyle) => {
      const defaultStyle = {
        buttonContainer: { backgroundColor: 'white' },
        buttonText: { color: 'red' },
      } as InAppMessageComponentBaseStyle;
      const styleParams = { defaultStyle, messageStyle, overrideStyle: null };

      const output = getComponentButtonStyle({
        styleParams,
        buttonType: 'primaryButton',
      });

      const buttonContainerStyle = (output as ResolveContainerStyle).container({
        pressed: true,
      });

      expect(buttonContainerStyle).toStrictEqual([
        pressedOpacity,
        { backgroundColor: 'white' },
        EMPTY_STYLE,
        EMPTY_STYLE,
      ]);
      expect(output.text).toStrictEqual([
        { color: 'red' },
        EMPTY_STYLE,
        EMPTY_STYLE,
      ]);
    }
  );

  describe('button container style', () => {
    it('returns unpressed button container style when press event is not provided', () => {
      const defaultStyle = {
        buttonContainer: { backgroundColor: 'white' },
        buttonText: {},
      } as InAppMessageComponentBaseStyle;
      const styleParams = {
        defaultStyle,
        messageStyle: null,
        overrideStyle: null,
      };

      const output = getComponentButtonStyle({
        styleParams,
        buttonType: 'primaryButton',
      });

      const buttonContainerStyle = (
        output as ResolveContainerStyle
      ).container();

      const expectedButtonContainerStyle = [
        EMPTY_STYLE,
        { backgroundColor: 'white' },
        EMPTY_STYLE,
        EMPTY_STYLE,
      ];

      expect(buttonContainerStyle).toStrictEqual(expectedButtonContainerStyle);
    });

    it('correctly evaluates button container override style when it is a function', () => {
      const pressedStyle = { backgroundColor: 'seafoam' };
      const unpressedStyle = { backgroundColor: 'fuschia' };
      const overrideStyle = {
        primaryButton: {
          container: ({ pressed }) => (pressed ? pressedStyle : unpressedStyle),
        },
      };
      const styleParams = {
        defaultStyle: null,
        messageStyle: null,
        overrideStyle,
      };

      const output = getComponentButtonStyle({
        styleParams,
        buttonType: 'primaryButton',
      });

      const containerPressedStyle = (output as ResolveContainerStyle).container(
        { pressed: true }
      );
      const containerUnpressedStyle = (
        output as ResolveContainerStyle
      ).container({ pressed: false });

      const expectedContainerPressedStyle = [
        pressedOpacity,
        EMPTY_STYLE,
        EMPTY_STYLE,
        pressedStyle,
      ];
      const expectedContainerUnressedStyle = [
        EMPTY_STYLE,
        EMPTY_STYLE,
        EMPTY_STYLE,
        unpressedStyle,
      ];

      expect(containerPressedStyle).toStrictEqual(
        expectedContainerPressedStyle
      );
      expect(containerUnpressedStyle).toStrictEqual(
        expectedContainerUnressedStyle
      );
    });
  });
});

describe('getContainerAndWrapperStyle', () => {
  it('returns the expected output for a banner component in the happy path', () => {
    const defaultStyle = {
      container: { backgroundColor: 'red' },
      componentWrapper: { opacity: 0.4 },
    } as InAppMessageComponentBaseStyle;
    const messageStyle = { container: { backgroundColor: 'teal' } };
    const overrideStyle = { container: { backgroundColor: 'pink' } };

    const output = getContainerAndWrapperStyle({
      layout: 'TOP_BANNER',
      styleParams: { defaultStyle, messageStyle, overrideStyle },
    });

    const expectedContainerStyle = [
      { backgroundColor: 'red' },
      { backgroundColor: 'teal' },
      { backgroundColor: 'pink' },
    ];

    const expectedWrapperStyle = { opacity: 0.4 };

    expect(output.container).toStrictEqual(expectedContainerStyle);
    expect(output.componentWrapper).toStrictEqual(expectedWrapperStyle);
  });

  it('returns the expected output for a non-banner component in the happy path', () => {
    const defaultStyle = {
      container: { backgroundColor: 'red' },
      componentWrapper: { opacity: 0.4 },
    } as InAppMessageComponentBaseStyle;
    const messageStyle = { container: { backgroundColor: 'teal' } };
    const overrideStyle = { container: { backgroundColor: 'pink' } };

    const output = getContainerAndWrapperStyle({
      layout: 'CAROUSEL',
      styleParams: { defaultStyle, messageStyle, overrideStyle },
    });

    const expectedContainerStyle = [{}, {}, {}];

    const expectedWrapperStyle = [
      { opacity: 0.4 },
      { backgroundColor: 'red' },
      { backgroundColor: 'teal' },
      { backgroundColor: 'pink' },
    ];

    expect(output.container).toStrictEqual(expectedContainerStyle);
    expect(output.componentWrapper).toStrictEqual(expectedWrapperStyle);
  });

  it('correctly handles a style array passed as the argument of overrideStyle.container', () => {
    const defaultStyle = {
      container: { backgroundColor: 'red' },
      componentWrapper: { opacity: 0.4 },
    } as InAppMessageComponentBaseStyle;
    const messageStyle = { container: { backgroundColor: 'teal' } };
    const overrideStyle = {
      container: [{ backgroundColor: 'pink' }, { flex: 5 }],
    };

    const output = getContainerAndWrapperStyle({
      layout: 'CAROUSEL',
      styleParams: { defaultStyle, messageStyle, overrideStyle },
    });

    const expectedContainerStyle = [EMPTY_STYLE, EMPTY_STYLE, { flex: 5 }];

    const expectedWrapperStyle = [
      { opacity: 0.4 },
      { backgroundColor: 'red' },
      { backgroundColor: 'teal' },
      { backgroundColor: 'pink' },
    ];

    expect(output.container).toStrictEqual(expectedContainerStyle);
    expect(output.componentWrapper).toStrictEqual(expectedWrapperStyle);
  });

  it('returns the expected output for a banner component with null style arguments', () => {
    const defaultStyle: StyleParams['defaultStyle'] = null;
    const messageStyle: StyleParams['messageStyle'] = null;
    const overrideStyle: StyleParams['overrideStyle'] = null;

    const output = getContainerAndWrapperStyle({
      layout: 'BOTTOM_BANNER',
      styleParams: { defaultStyle, messageStyle, overrideStyle },
    });

    const expectedContainerStyle = [EMPTY_STYLE, EMPTY_STYLE, EMPTY_STYLE];

    const expectedWrapperStyle = EMPTY_STYLE;

    expect(output.container).toStrictEqual(expectedContainerStyle);
    expect(output.componentWrapper).toStrictEqual(expectedWrapperStyle);
  });

  it('returns the expected output for a non-banner component with empty style arguments', () => {
    const defaultStyle: StyleParams['defaultStyle'] = null;
    const messageStyle: StyleParams['messageStyle'] = null;
    const overrideStyle: StyleParams['overrideStyle'] = null;

    const output = getContainerAndWrapperStyle({
      layout: 'MODAL',
      styleParams: { defaultStyle, messageStyle, overrideStyle },
    });

    const expectedContainerStyle = [EMPTY_STYLE, EMPTY_STYLE, EMPTY_STYLE];

    const expectedWrapperStyle = [
      EMPTY_STYLE,
      EMPTY_STYLE,
      EMPTY_STYLE,
      EMPTY_STYLE,
    ];

    expect(output.container).toStrictEqual(expectedContainerStyle);
    expect(output.componentWrapper).toStrictEqual(expectedWrapperStyle);
  });
});

describe('getMessageStyle', () => {
  it('returns the expected output in the happy path', () => {
    const output = getMessageStyle({
      body: { style: { textAlign: 'left' as InAppMessageTextAlign } },
      container: { style: { backgroundColor: 'lightgray', borderRadius: 2 } },
      header: { style: { textAlign: 'center' as InAppMessageTextAlign } },
      primaryButton: { style: { backgroundColor: 'salmon', color: 'olive' } },
      secondaryButton: { style: { backgroundColor: 'sand', color: 'peru' } },
    } as InAppMessageComponentBaseProps);

    expect(output).toMatchSnapshot();
  });

  it('returns the expected output when given empty style values', () => {
    const output = getMessageStyle({
      body: { style: null },
      container: { style: null },
      header: { style: null },
      primaryButton: { style: null },
      secondaryButton: { style: null },
    } as InAppMessageComponentBaseProps);

    expect(output).toMatchSnapshot();
  });
});

describe('getMessageStyleProps', () => {
  it('returns the expected output in the happy path', () => {
    const defaultStyle = {
      body: { color: 'fuschia' },
      buttonContainer: { backgroundColor: 'chartreuse' },
      buttonText: { color: 'pink' },
      buttonsContainer: { backgroundColor: 'teal' },
      componentWrapper: { backgroundColor: 'gray' },
      contentContainer: { backgroundColor: 'lightblue' },
      container: { backgroundColor: 'red', borderRadius: 1 },
      header: { backgroundColor: 'purple' },
      iconButton: { backgroundColor: 'blue' },
      image: { backgroundColor: 'yellow' },
      imageContainer: { backgroundColor: 'green' },
      textContainer: { backgroundColor: 'antiquewhite' },
    };

    const messageStyle: StyleParams['messageStyle'] = {
      body: { textAlign: 'left' as InAppMessageTextAlign },
      container: { backgroundColor: 'lightgray', borderRadius: 2 },
      header: { textAlign: 'center' as InAppMessageTextAlign },
      primaryButton: { backgroundColor: 'salmon', color: 'olive' },
      secondaryButton: { backgroundColor: 'sand', color: 'peru' },
    };

    const overrideStyle = {
      body: { color: 'white' },
      closeIconButton: { backgroundColor: 'turquoise' },
      closeIconColor: 'darkcyan',
      container: { backgroundColor: 'lawngreen', borderRadius: 3 },
      header: { backgroundColor: 'lightpink' },
      image: { backgroundColor: 'royalblue' },
      primaryButton: {
        container: { backgroundColor: 'seagreen' },
        text: { color: 'black' },
      },
      secondaryButton: {
        container: { backgroundColor: 'sienna' },
        text: { color: 'orchid' },
      },
    };

    const output = getMessageStyleProps({
      layout: 'FULL_SCREEN',
      styleParams: { defaultStyle, messageStyle, overrideStyle },
    });

    expect(output).toMatchSnapshot();
  });

  it('returns the expected output when provided null style params', () => {
    const defaultStyle: StyleParams['defaultStyle'] = null;
    const messageStyle: StyleParams['messageStyle'] = null;
    const overrideStyle: StyleParams['overrideStyle'] = null;

    const output = getMessageStyleProps({
      layout: 'MODAL',
      styleParams: { defaultStyle, messageStyle, overrideStyle },
    });

    expect(output).toMatchSnapshot();
  });
});
