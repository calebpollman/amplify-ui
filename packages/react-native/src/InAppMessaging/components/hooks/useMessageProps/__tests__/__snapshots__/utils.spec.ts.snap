// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMessageStyle returns the expected output in the happy path 1`] = `
Object {
  "body": Object {
    "textAlign": "left",
  },
  "container": Object {
    "backgroundColor": "lightgray",
    "borderRadius": 2,
  },
  "header": Object {
    "textAlign": "center",
  },
  "primaryButton": Object {
    "backgroundColor": "salmon",
    "color": "olive",
  },
  "secondaryButton": Object {
    "backgroundColor": "sand",
    "color": "peru",
  },
}
`;

exports[`getMessageStyle returns the expected output when given empty style values 1`] = `
Object {
  "body": Object {},
  "container": Object {},
  "header": Object {},
  "primaryButton": Object {},
  "secondaryButton": Object {},
}
`;

exports[`getMessageStyleProps returns the expected output in the happy path 1`] = `
Object {
  "body": Array [
    Object {
      "color": "fuschia",
    },
    Object {
      "textAlign": "left",
    },
    Object {
      "color": "white",
    },
  ],
  "buttonsContainer": Object {
    "backgroundColor": "teal",
  },
  "componentWrapper": Array [
    Object {
      "backgroundColor": "gray",
    },
    Object {
      "backgroundColor": "red",
    },
    Object {
      "backgroundColor": "lightgray",
    },
    Object {
      "backgroundColor": "lawngreen",
    },
  ],
  "container": Array [
    Object {
      "borderRadius": 1,
    },
    Object {
      "borderRadius": 2,
    },
    Object {
      "borderRadius": 3,
    },
  ],
  "contentContainer": Object {
    "backgroundColor": "lightblue",
  },
  "header": Array [
    Object {
      "backgroundColor": "purple",
    },
    Object {
      "textAlign": "center",
    },
    Object {
      "backgroundColor": "lightpink",
    },
  ],
  "iconButton": Object {
    "container": Array [
      Object {
        "backgroundColor": "blue",
      },
      Object {
        "backgroundColor": "turquoise",
      },
    ],
    "iconColor": "darkcyan",
  },
  "image": Array [
    Object {
      "backgroundColor": "yellow",
    },
    Object {
      "backgroundColor": "royalblue",
    },
  ],
  "imageContainer": Object {
    "backgroundColor": "green",
  },
  "primaryButton": Object {
    "container": [Function],
    "text": Array [
      Object {
        "color": "pink",
      },
      Object {
        "color": "olive",
      },
      Object {
        "color": "black",
      },
    ],
  },
  "secondaryButton": Object {
    "container": [Function],
    "text": Array [
      Object {
        "color": "pink",
      },
      Object {
        "color": "peru",
      },
      Object {
        "color": "orchid",
      },
    ],
  },
  "textContainer": Object {
    "backgroundColor": "antiquewhite",
  },
}
`;

exports[`getMessageStyleProps returns the expected output when provided null style params 1`] = `
Object {
  "body": Array [
    undefined,
    undefined,
    undefined,
  ],
  "buttonsContainer": undefined,
  "componentWrapper": Array [
    Object {},
    Object {},
    Object {},
    Object {},
  ],
  "container": Array [
    Object {},
    Object {},
    Object {},
  ],
  "contentContainer": undefined,
  "header": Array [
    undefined,
    undefined,
    undefined,
  ],
  "iconButton": Object {
    "container": Array [
      undefined,
      undefined,
    ],
    "iconColor": undefined,
  },
  "image": Array [
    undefined,
    undefined,
  ],
  "imageContainer": undefined,
  "primaryButton": Object {
    "container": [Function],
    "text": Array [
      Object {},
      Object {},
      Object {},
    ],
  },
  "secondaryButton": Object {
    "container": [Function],
    "text": Array [
      Object {},
      Object {},
      Object {},
    ],
  },
  "textContainer": undefined,
}
`;
